druid/examples/image.rs

```diff
@@ -12,44 +12,319 @@
+    }
+
+    fn update(&mut self, ctx: &mut UpdateCtx, old_data: &AppState, data: &AppState, env: &Env) {
+        if !old_data.params.same(&data.params) {
+            self.rebuild_inner(data);
+            ctx.children_changed();
+        } else {
+            self.inner.update(ctx, old_data, data, env);
+        }
+    }
+
+    fn layout(
+        &mut self,
+        ctx: &mut LayoutCtx,
+        bc: &BoxConstraints,
+        data: &AppState,
+        env: &Env,
+    ) -> Size {
+        self.inner.layout(ctx, bc, data, env)
+    }
+
+    fn paint(&mut self, ctx: &mut PaintCtx, data: &AppState, env: &Env) {
+        self.inner.paint(ctx, data, env)
+    }
+
+    fn id(&self) -> Option<WidgetId> {
+        self.inner.id()
+    }
+}
+
+fn make_control_row() -> impl Widget<AppState> {
+    Flex::row()
+        .cross_axis_alignment(CrossAxisAlignment::Start)
+        .with_child(
+            Flex::column()
+                .cross_axis_alignment(CrossAxisAlignment::Start)
+                .with_child(Label::new("FillStrat:"))
+                .with_default_spacer()
+                .with_child(RadioGroup::new(FILL_STRAT_OPTIONS.to_vec()).lens(Params::fill_strat)),
+        )
+        .with_default_spacer()
+        .with_child(
+            Flex::column()
+                .cross_axis_alignment(CrossAxisAlignment::Start)
+                .with_child(Label::new("interpolation mode:"))
+                .with_default_spacer()
+                .with_child(Checkbox::new("set interpolation mode").lens(Params::interpolate))
+                .with_default_spacer()
+                .with_child(
+                    RadioGroup::new(INTERPOLATION_MODE_OPTIONS.to_vec())
+                        .lens(Params::interpolation_mode),
+                ),
+        )
+        .with_default_spacer()
+        .with_child(make_width())
+        .with_default_spacer()
+        .with_child(make_height())
+        .with_default_spacer()
+        .with_child(
+            Flex::column()
+                .cross_axis_alignment(CrossAxisAlignment::Start)
+                .with_child(Label::new("Misc:"))
+                .with_default_spacer()
+                .with_child(Checkbox::new("Debug layout").lens(Params::debug_layout))
+                .with_default_spacer()
+                .with_child(Checkbox::new("Fix width").lens(Params::fix_width))
+                .with_default_spacer()
+                .with_child(Checkbox::new("Fix height").lens(Params::fix_height)),
+        )
+        .padding(10.0)
+        .border(Color::grey(0.6), 2.0)
+        .rounded(5.0)
+        .lens(AppState::params)
+}
+
+fn make_width() -> impl Widget<Params> {
+    Flex::column()
+        .cross_axis_alignment(CrossAxisAlignment::Start)
+        .with_child(Label::new("width:"))
+        .with_default_spacer()
+        .with_child(
+            Flex::row()
+                .with_child(
+                    TextBox::new()
+                        .parse()
```

Comment:
I'd like to not use `parse()` in new code; instead you should call `with_formatter` and pass in a `ParseFormatter`.